### PR TITLE
wip: gstin status

### PR DESCRIPTION
**Pull request by DaizyModi**

### CODE REVIEW

File: india_compliance/gst_india/doctype/gstin_detail/gstin_detail.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+   "email": 1,
+   "export": 1,
+   "print": 1,
+   "read": 1,
+   "report": 1,
+   "role": "System Manager",
+   "share": 1,
+   "write": 1
+  }
+ ],
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gstin_detail/gstin_detail.json]: Set appropriate permissions

File: india_compliance/gst_india/doctype/gstin_detail/gstin_detail.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+   "label": "Status",
+   "read_only": 1
+  },
+  {
+   "fieldname": "last_updated_on",
+   "fieldtype": "Datetime",
+   "label": "Last Updated On",
+   "read_only": 1
+  }
+ ],
```
Comment (vorasmit) [inline india_compliance/gst_india/doctype/gstin_detail/gstin_detail.json]: Appropriate List view

File: india_compliance/gst_india/client_scripts/party.js [hunk 3]
```diff
@@ -105,3 +105,15 @@ function show_overseas_disabled_warning(doctype) {
...
     });
 }
+
+function show_gstin_status_in_description(doctype) {
+    frappe.ui.form.on(doctype, {
+        onload(frm) {
+            if (!frm.doc.gstin) return;
+
+            frappe.db.get_value("GSTIN Detail", { 'name': frm.doc.gstin }, "status", (r) => {
+                frm.get_field('gstin').set_description(__('{0}', [r.status || "No status found"]));
```
Comment (vorasmit) [inline india_compliance/gst_india/client_scripts/party.js]: Default `None` or no string

File: india_compliance/gst_india/overrides/transaction.py [hunk 4]
```diff
@@ -660,3 +660,16 @@ def validate_transaction(doc, method=None):
 
     valid_accounts = validate_gst_accounts(doc, is_sales_transaction) or ()
     update_taxable_values(doc, valid_accounts)
+
+
+def validate_gstin_status(gstin):
+    if not gstin:
+        return
+
+    gstin_status = frappe.get_cached_value("GSTIN Detail", gstin, "status")
```
Comment (vorasmit) [inline india_compliance/gst_india/overrides/transaction.py]: We may not like to cache this

File: india_compliance/gst_india/overrides/transaction.py [hunk 5]
```diff
@@ -660,3 +660,16 @@ def validate_transaction(doc, method=None):
...
+
+    if gstin_status == "Active":
```
Comment (vorasmit) [inline india_compliance/gst_india/overrides/transaction.py]:
```suggestion
    if gstin_status != "Active":
```

File: india_compliance/gst_india/overrides/transaction.py [hunk 6]
```diff
@@ -660,3 +660,16 @@ def validate_transaction(doc, method=None):
...
+        frappe.throw(
+            _(f"Billing Address GSTIN is {gstin_status}"),
+            title=_("Invalid Billing Address GSTIN"),
```
Comment (vorasmit) [inline india_compliance/gst_india/overrides/transaction.py]:
```suggestion
            title=_("{0} Billing Address GSTIN".format(gstin_status)),
```

File: india_compliance/gst_india/overrides/transaction.py [hunk 7]
```diff
@@ -660,3 +660,16 @@ def validate_transaction(doc, method=None):
...
+
+def validate_gstin_status(gstin):
+    if not gstin:
+        return
+
+    gstin_status = frappe.get_cached_value("GSTIN Detail", gstin, "status")
+
+    if gstin_status == "Active":
+        frappe.throw(
+            _(f"Billing Address GSTIN is {gstin_status}"),
```
Comment (vorasmit) [inline india_compliance/gst_india/overrides/transaction.py]: cannot use f string with translate

File: india_compliance/gst_india/overrides/sales_invoice.py [hunk 8]
```diff
@@ -55,6 +56,7 @@ def validate(doc, method=None):
     validate_mandatory_fields(doc, ("place_of_supply", "gst_category"))
     validate_fields_and_set_status_for_e_invoice(doc)
     validate_billing_address_gstin(doc)
+    validate_gstin_status(doc.billing_address_gstin)
```
Comment (vorasmit) [inline india_compliance/gst_india/overrides/sales_invoice.py]: Instead add this to `validate_transaction` in transaction.py

**Comment by sagarvora** [comment]
reopen when ready.

**Comment by DaizyModi** [comment]
Continues in PR #735

**Pull request closed without merging**